website/common/script/libs/getOfficialPinnedItems.js

```diff
@@ -14,14 +30,16 @@ export default function getOfficialPinnedItems (user) {
 
     // pinnedSets == current seasonal class set are always gold purchaseable
 
-    flatGearArray
-      .filter(gear => user.items.gear.owned[gear.key] === undefined && gear.set === setToAdd)
-      .forEach(gear => {
-        officialItemsArray.push({
+    const gearsBySet = flatGearArray[setToAdd];
+    const gears = gearsBySet.map(gear => {
```

Comment:
we could squeeze a little more performance out of it with using a `for (const gear of gearsBySet)` and just adding again with push

that way there is no need to `.map` (which creates a new array at the end) which could be just used to add to `officialItemsArray` again - yay performance 😬

Reply:
for of is one of the slowest loops we have in js
i'm using the fastest now, classical for